be/src/storage/lake/general_tablet_writer.cpp

```diff
@@ -0,0 +1,87 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#include "storage/lake/general_tablet_writer.h"
+
+#include <fmt/format.h>
+
+#include "column/chunk.h"
+#include "common/config.h"
+#include "fs/fs_util.h"
+#include "storage/rowset/segment_writer.h"
+#include "util/uid_util.h"
+
+namespace starrocks::lake {
+
+GeneralTabletWriter::GeneralTabletWriter(Tablet tablet) : _tablet(std::move(tablet)) {}
+
+GeneralTabletWriter::~GeneralTabletWriter() {}
+
+Status GeneralTabletWriter::open() {
+    DCHECK(_schema == nullptr);
+    ASSIGN_OR_RETURN(_schema, _tablet.get_schema());
+    return Status::OK();
+}
+
+Status GeneralTabletWriter::write(const starrocks::vectorized::Chunk& data) {
+    if (_seg_writer == nullptr || _seg_writer->estimate_segment_size() >= config::max_segment_file_size ||
+        _seg_writer->num_rows_written() + data.num_rows() >= INT32_MAX /*TODO: configurable*/) {
+        RETURN_IF_ERROR(flush_segment_writer());
+        RETURN_IF_ERROR(reset_segment_writer());
+    }
+    RETURN_IF_ERROR(_seg_writer->append_chunk(data));
+    _num_rows += data.num_rows();
+    return Status::OK();
+}
+
+Status GeneralTabletWriter::flush() {
+    return flush_segment_writer();
+}
+
+Status GeneralTabletWriter::finish() {
+    RETURN_IF_ERROR(flush_segment_writer());
+    _finished = true;
+    return Status::OK();
+}
+
+void GeneralTabletWriter::close() {
+    if (!_finished) {
+        // Delete files
+        auto group = _tablet.group();
+        auto maybe_fs = FileSystem::CreateSharedFromString(group);
+        if (!maybe_fs.ok()) {
+            return;
+        }
+        auto fs = std::move(maybe_fs).value();
+        for (auto&& name : _files) {
```

Comment:
why use rvalue reference.

Reply:
This is not an rvalue reference but a universal reference or, forwarding reference.

Reply:
I replaced `auto&&` with `const auto&` for easier understanding，PTAL.